assets/src/stories-editor/blocks/amp-story-page/animation-settings.js

```diff
@@ -27,24 +28,35 @@ import stopIcon from '../../../../images/stories-editor/stop.svg';
  * @param {string} props.clientId Client ID.
  */
 const AnimationSettings = ( { clientId } ) => {
-	const { playAnimation, stopAnimation } = useDispatch( 'amp/story' );
-	const onAnimationStart = () => playAnimation( clientId );
-	const onAnimationStop = () => stopAnimation( clientId );
-	const isPlayingAnimations = useSelect( ( select ) => {
-		const { isPlayingAnimation } = select( 'amp/story' );
-		return isPlayingAnimation( clientId );
-	} );
-	const animatedBlocks = useSelect( ( select ) => {
+	const {
+		isPlayingAnimation,
+		animatedBlocks,
+	} = useSelect( ( select ) => {
 		const { getBlock } = select( 'core/block-editor' );
 		const { getAnimatedBlocksPerPage } = select( 'amp/story' );
-		return getAnimatedBlocksPerPage( clientId ).filter( ( { id, animationType } ) => animationType && getBlock( id ) );
-	} );
+
+		return {
+			isPlayingAnimation: select( 'amp/story' ).isPlayingAnimation( clientId ),
+			animatedBlocks: getAnimatedBlocksPerPage( clientId ).filter( ( { id, animationType } ) => animationType && getBlock( id ) ),
+		};
+	}, [ clientId ] );
+
+	const { playAnimation, stopAnimation } = useDispatch( 'amp/story' );
+
+	const onAnimationStart = useCallback(
+		() => playAnimation( clientId ),
+		[ clientId, playAnimation ]
```

Comment:
^ Here, too, we don't really need to track `playAnimation`

Reply:
In theory, we wouldn't need to add `playAnimation` here since `clientId` already has been added, based on https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

However, I guess there's no harm doing it either, as you mentioned. It's just weird to read for me (will get used to it).